### DISCUSSION TITLE
Search for scripts from multiple sites

### DISCUSSION THREAD

**Discussion by jojoPL1987** [violentmonkey/violentmonkey#1261] [Ideas]
In the pop-up there is an option "Find scripts for this page". After selecting it using the greasyfork.org website, we are looking for a script for the open page (when we selected that option with the pop-up). But why is the search done with used of Greasy Fork and not another site?

**My suggestion is to add the ability to change this page and even add new scripted searching pages.**

In the browser I use, it is possible to add my own search engines. You simply paste the address from the search engine into the appropriate field, but the searched phrase changes to "%s".

If we adopted such a rule, adding Grasy Fork as a new page to search would require entering the following address:
`https://greasyfork.org/scripts/by-site/%s`

And on this basis, the user could add their own search engines. There is only a question: the search for scripts would be on all added pages immediately or maybe add a sub-list with all search engines to the pop-up so that the user could choose a specific one?

Such a novelty can bring one more benefit. Namely, you can edit the default GreasyFork search engine. My native language is not English (which can be seen from the imperfect translation;)) and when searching for scripts in Grasy Fork, I often get the message "no scripts found" by default selection of my language. And then I have to click on "Show results for all languages".

After introducing such a novelty, I could additionally edit the Greasy Fork search engine to the following address:
`https://greasyfork.org/scripts/by-site/%s?filter_locale=0`
So I would have scripts for all languages ​​displayed right away.

**Comment by tophf** [comment]
I like the idea of one customizable search engine in `Advanced` settings.

**Reply by jojoPL1987** [reply]
I myself have 3 sources of scripts and I would just prefer to search for scripts from the options in the pop-up on all of these 3 pages. But of course, if someone wants, he can have 1 page set.
